doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,104 @@
+# Single executable applications
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental: This feature is being designed and will change.
+
+<!-- source_link=lib/internal/main/single_executable_application.js -->
+
+This feature allows the distribution of a Node.js application conveniently to a
+system that does not have Node.js installed.
+
+Node.js supports the creation of [single executable applications][] by allowing
+the injection of a JavaScript file into the `node` binary. During start up, the
+program checks if anything has been injected. If the script is found, it
+executes its contents. Otherwise it operates like plain Node.js.
+
+A bundled JavaScript file can be turned into a single executable application
+with any tool which can inject resources into the `node` binary.
+
+One such tool is [postject][]:
+
+```console
+$ cat hello.js
+console.log(`Hello, ${process.argv[2]}!`);
+$ cp $(command -v node) hello
+# On systems other than macOS:
+$ npx postject hello NODE_JS_CODE hello.js \
+    --sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2
```

Comment:
What is `--sentinel-fuse NODE_JS_FUSE_fce680ab2cc467b6e072b8b5df1996b2`? How is this value generated?

Reply:
The string is a constant I generated by appending some random hex characters after `NODE_JS_FUSE_`. The flag allows you to pass the [fuse](https://www.electronjs.org/docs/latest/tutorial/fuses) which stores a byte that is used by the `node` binary to detect if something has been injected in a zero-cost way.

Reply:
I think this should be explained. If it's a constant, why can't `postject` know about it?

Reply:
> I think this should be explained.

The `Single executable application creation process` section of the doc already has a link to https://www.electronjs.org/docs/latest/tutorial/fuses. Could you tell me which part of fuses I should explain in the docs?

>  If it's a constant, why can't postject know about it?

`postject` has been built in such a way that it would work with other programs also, like Deno, Python, etc. We provide the option for other projects to set their own sentinel strings based on their preference, so in this case by "constant", I mean it's a constant for the Node.js project.

Reply:
These steps are absolutely not clear. I think an explanation to what you ask people to do is necessary, otherwise the snippet is not clear.

Reply:
@mcollina I've pushed a commit to explain each step. Does this address your concern?